rmw_cyclonedds_cpp/src/rmw_get_topic_endpoint_info.cpp

```diff
@@ -12,6 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "rmw_cyclonedds_cpp/rmw_version_test.hpp"
+
+#if RMW_VERSION_GTE(0, 8, 1)
```

Comment:
The missing functions weren't part of release `0.8.1`, so `0.8.1` shouldn't be included. The problem is that we usually bump the version when releasing and not when developing.

I see that `Dashing` is using `master` branch ([1](https://github.com/ros2/ros2/blob/07c28ab41435154f6c1cee2dac33a990e97e9873/ros2.repos#L245)), while eloquent is using it's own branch ([2](https://github.com/ros2/ros2/blob/eloquent/ros2.repos)). Why the incosistence?
If there weren't such inconsistency, including the new functions in `0.8.1` would be a problem.

Reply:
Ok, I saw this comment about development versions https://github.com/ros2/rmw/issues/188#issuecomment-540233493, and also this:
https://github.com/ros2/rmw/blob/a3774531e9b7289574501275aafbc3f93120360e/rmw/CMakeLists.txt#L66.

I'm not pretty sure about how it should be used, but I think that should be considered in this PR.

Reply:
@ivanpauno, thanks for looking into it. The ``0.8.1`` comes straight from the Foxy build settings. I guess that one should have been bumped then.

Unless I am mistaken, I'm the one who asked for the rmw versioning stuff to work nicely so that I could have a single source for multiple versions without having to deal with backporting things. That used to be perfectly viable, but I have a feeling https://github.com/ros2/design/pull/250 is going to happen, and that will be too big a change to handle with a bit of conditional compilation. Therefore, it seems multiple branches are likely to become unavoidable. 

Since it turns out a branch had already been created for eloquent, I guess we now should retroactively give dashing a branch as well. Then this PR can simply be closed.

Reply:
> @ivanpauno, thanks for looking into it. The 0.8.1 comes straight from the Foxy build settings. I guess that one should have been bumped then.

mmm, it wasn't bumped. The `eloquent` version of `rmw` is indicating `0.8.1`: https://github.com/ros2/rmw/blob/eloquent/rmw/package.xml.
I'm unsure how the versioning was supposed to be used.
@wjwwood @dirk-thomas can you clarify?

> Unless I am mistaken, I'm the one who asked for the rmw versioning stuff to work nicely so that I could have a single source for multiple versions without having to deal with backporting things

Yes, that's true. I agree that mantaining similar versions from the same source is desired.

> Since it turns out a branch had already been created for eloquent, I guess we now should retroactively give dashing a branch as well. Then this PR can simply be closed.

I guess that probably was an error, if it's easier to mantain `eloquent` and `dashing` from the same branch it would be reasonable to merge them again.

> That used to be perfectly viable, but I have a feeling ros2/design#250 is going to happen, and that will be too big a change to handle with a bit of conditional compilation. Therefore, it seems multiple branches are likely to become unavoidable.

Yeah, I agree that after that change it will be easier to branch that to mantain all the versions from the same branch.

Reply:
> I'm unsure how the versioning was supposed to be used.

Whenever the API is broken in `rmw` it should declare that by bumping the dev version to the next upcoming version number: e.g. https://github.com/ros2/rmw/commit/688e53a976970de837c39c11e965b9d56c438e07

Reply:
Yeah, we should have updated it when merging that. I forgot to do so. It's tricky to remember to do that when pull requests span releases, but we'll just have to be more diligent about it.

Reply:
I'm working on a pull request to fix it.

Reply:
https://github.com/ros2/rmw/pull/194

Reply:
Thanks for the PR @wjwwood !
@eboasson it should be greater or equal than `0.8.2` then, LGTM with that update.